.github/workflows/ci.yaml

```
@@ -0,0 +1,80 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: sssd charm tests
+on:
+  workflow_call:
+  pull_request:
+
+jobs:
+  inclusive-naming-check:
+    name: Inclusive naming check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: woke
+        uses: get-woke/woke-action@v0
+        with:
+          # Cause the check to fail on any broke rules
+          fail-on-error: true
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run linters
+        run: tox -e lint
+
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run tests
+        run: tox -e unit
+
+  functional-test:
+    name: Functional tests
+    runs-on: ubuntu-latest
+    needs:
+      - inclusive-naming-check
+      - unit-test
+      - lint
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: python -m pip install tox
+      - name: Setup LXD
+        uses: canonical/setup-lxd@ea57509243d3cf39f8ab926e021bb353947b01b5
```

So the LXD action actually has point releases now so you should not need to use a specific commit hash unless there is a specific reason you need to. You can change the action version to the following:

```suggestion
        uses: canonical/setup-lxd@v0.1.0
```

---

Cool, updated!